service/controller/internal/vmsku/vmsku.go

```diff
@@ -0,0 +1,107 @@
+package vmsku
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/azure-operator/v4/client"
+)
+
+const (
+	// CapabilitySupported is the value returned by this API from Azure when the capability is supported
+	CapabilitySupported = "True"
+
+	CapabilityAcceleratedNetworking = "AcceleratedNetworkingEnabled"
+)
+
+type Config struct {
+	ClientFactory *client.Factory
+	Location      string
+	Logger        micrologger.Logger
+}
+
+type Interface struct {
+	clientFactory *client.Factory
+	location      string
+	skus          map[string]*compute.ResourceSku
+	logger        micrologger.Logger
+}
+
+func New(config Config) (*Interface, error) {
+	if config.ClientFactory == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ClientFactory must not be empty", config)
+	}
+	if config.Location == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Location must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	return &Interface{
+		clientFactory: config.ClientFactory,
+		location:      config.Location,
+		logger:        config.Logger,
+	}, nil
+}
+
+func (v *Interface) HasCapability(ctx context.Context, vmType string, name string) (bool, error) {
+	if len(v.skus) == 0 {
+		err := v.initCache(ctx)
+		if err != nil {
+			return false, microerror.Mask(err)
+		}
+	}
+	vmsku, found := v.skus[vmType]
+	if !found {
+		return false, microerror.Maskf(skuNotFoundError, vmType)
+	}
+	if vmsku.Capabilities != nil {
+		for _, capability := range *vmsku.Capabilities {
+			if capability.Name != nil && *capability.Name == name {
+				if capability.Value != nil && strings.EqualFold(*capability.Value, string(CapabilitySupported)) {
+					return true, nil
+				}
+			}
+		}
+	}
+	return false, nil
+}
+
+func (v *Interface) initCache(ctx context.Context) error {
```

Comment:
I think this should be somehow guarded with a mutex to only be initialized once. Alternatively one could also utilize [`sync.Once`](https://golang.org/pkg/sync/#Once).

Reply:
done with a mutex